File: typescript/tests/wire/harnesses.test.ts

import * as Sikaru from "../../src/api/index";
import { SikaruApi } from "../../src/Client";
import { mockServerPool } from "../mock-server/MockServerPool";

describe("HarnessesClient", () => {
    
    test("improvement_options (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/harnesses/harness_id/improvement-options").respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.harnesses.improvementOptions("project_id", "harness_id");
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("improvement_options (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/harnesses/harness_id/improvement-options").respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.harnesses.improvementOptions("project_id", "harness_id")
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("list_improvements (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/harnesses/harness_id/improvements").respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.harnesses.listImprovements("project_id", "harness_id");
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("list_improvements (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/harnesses/harness_id/improvements").respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.harnesses.listImprovements("project_id", "harness_id")
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("start_improvement (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        const rawRequestBody = { "idempotency_key" : "idempotency_key" };
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/harnesses/harness_id/improvements").jsonBody(rawRequestBody)
                .respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.harnesses.startImprovement("project_id", "harness_id", {
    idempotency_key: "idempotency_key"
});
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("start_improvement (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        const rawRequestBody = { "idempotency_key" : "x" };
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/harnesses/harness_id/improvements").jsonBody(rawRequestBody)
                .respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.harnesses.startImprovement("project_id", "harness_id", {
    idempotency_key: "x"
})
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("get_improvement (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/harnesses/harness_id/improvements/job_id").respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.harnesses.getImprovement("project_id", "harness_id", "job_id");
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("get_improvement (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/harnesses/harness_id/improvements/job_id").respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.harnesses.getImprovement("project_id", "harness_id", "job_id")
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("resume_improvement (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        const rawRequestBody = { };
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/harnesses/harness_id/improvements/job_id/resume").jsonBody(rawRequestBody)
                .respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.harnesses.resumeImprovement("project_id", "harness_id", "job_id", {});
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("resume_improvement (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        const rawRequestBody = null;
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/harnesses/harness_id/improvements/job_id/resume").jsonBody(rawRequestBody)
                .respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.harnesses.resumeImprovement("project_id", "harness_id", "job_id", null)
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("train_model_stub (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/harnesses/harness_id/training").respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.harnesses.trainModelStub("project_id", "harness_id");
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("train_model_stub (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/harnesses/harness_id/training").respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.harnesses.trainModelStub("project_id", "harness_id")
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
});

File: ruby/test/wire/harnesses_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class HarnessesWireTest < WireMockTestCase
  def setup
    super

    @client = Sikaru::Client.new(
      token: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_harnesses_get_invoice_budget_with_wiremock
    test_id = "harnesses.get_invoice_budget.0"

    @client.harnesses.get_invoice_budget(
      project_id: "project_id",
      harness_id: "harness_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "harnesses.get_invoice_budget.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/projects/project_id/harnesses/harness_id/budget/invoice",
      query_params: nil,
      expected: 1
    )
  end

  def test_harnesses_get_subscription_with_wiremock
    test_id = "harnesses.get_subscription.0"

    @client.harnesses.get_subscription(
      project_id: "project_id",
      harness_id: "harness_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "harnesses.get_subscription.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/projects/project_id/harnesses/harness_id/budget/subscription",
      query_params: nil,
      expected: 1
    )
  end

  def test_harnesses_subscribe_with_wiremock
    test_id = "harnesses.subscribe.0"

    @client.harnesses.subscribe(
      project_id: "project_id",
      harness_id: "harness_id",
      accepted_recurring_terms: true,
      idempotency_key: "idempotency_key",
      plan: "build",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "harnesses.subscribe.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/harnesses/harness_id/budget/subscription",
      query_params: nil,
      expected: 1
    )
  end

  def test_harnesses_cancel_subscription_with_wiremock
    test_id = "harnesses.cancel_subscription.0"

    @client.harnesses.cancel_subscription(
      project_id: "project_id",
      harness_id: "harness_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "harnesses.cancel_subscription.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/harnesses/harness_id/budget/subscription/cancel",
      query_params: nil,
      expected: 1
    )
  end

  def test_harnesses_improvement_options_with_wiremock
    test_id = "harnesses.improvement_options.0"

    @client.harnesses.improvement_options(
      project_id: "project_id",
      harness_id: "harness_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "harnesses.improvement_options.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/projects/project_id/harnesses/harness_id/improvement-options",
      query_params: nil,
      expected: 1
    )
  end

  def test_harnesses_list_improvements_with_wiremock
    test_id = "harnesses.list_improvements.0"

    @client.harnesses.list_improvements(
      project_id: "project_id",
      harness_id: "harness_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "harnesses.list_improvements.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/projects/project_id/harnesses/harness_id/improvements",
      query_params: nil,
      expected: 1
    )
  end

  def test_harnesses_start_improvement_with_wiremock
    test_id = "harnesses.start_improvement.0"

    @client.harnesses.start_improvement(
      project_id: "project_id",
      harness_id: "harness_id",
      idempotency_key: "idempotency_key",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "harnesses.start_improvement.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/harnesses/harness_id/improvements",
      query_params: nil,
      expected: 1
    )
  end

  def test_harnesses_get_improvement_with_wiremock
    test_id = "harnesses.get_improvement.0"

    @client.harnesses.get_improvement(
      project_id: "project_id",
      harness_id: "harness_id",
      job_id: "job_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "harnesses.get_improvement.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/projects/project_id/harnesses/harness_id/improvements/job_id",
      query_params: nil,
      expected: 1
    )
  end

  def test_harnesses_resume_improvement_with_wiremock
    test_id = "harnesses.resume_improvement.0"

    @client.harnesses.resume_improvement(
      project_id: "project_id",
      harness_id: "harness_id",
      job_id: "job_id",
      request: {},
      request_options: {
        additional_headers: {
          "X-Test-Id" => "harnesses.resume_improvement.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/harnesses/harness_id/improvements/job_id/resume",
      query_params: nil,
      expected: 1
    )
  end

  def test_harnesses_train_model_stub_with_wiremock
    test_id = "harnesses.train_model_stub.0"

    @client.harnesses.train_model_stub(
      project_id: "project_id",
      harness_id: "harness_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "harnesses.train_model_stub.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/harnesses/harness_id/training",
      query_params: nil,
      expected: 1
    )
  end
end
